ca_gazebo/src/virtual_wall_detector.cpp

```diff
@@ -0,0 +1,61 @@
+#include "ca_gazebo/virtual_wall_detector.h"
+
+#include <gazebo/common/Plugin.hh>
+
+
+namespace gazebo
+{
+  // Register this plugin with the simulator
+  GZ_REGISTER_MODEL_PLUGIN(GazeboVirtualWallDetector)
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Constructor
+  GazeboVirtualWallDetector::GazeboVirtualWallDetector():
+    nh_("virtual_wall_detector_plugin")
+{
+
+}
+  ////////////////////////////////////////////////////////////////////////////////
+  // Destructor
+  GazeboVirtualWallDetector::~GazeboVirtualWallDetector()
+  {
+  }
+
+  void GazeboVirtualWallDetector::Load(physics::ModelPtr _parent, sdf::ElementPtr _sdf)
+  {
+    // Make sure the ROS node for Gazebo has already been initialized
+    if (!ros::isInitialized())
+    {
+      ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load plugin. "
+        << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)");
+      return;
+    }
+    // Initialize the ros variables and gaebo variables
+    this->virtual_wall_publisher_ = nh_.advertise<std_msgs::Bool>("/filtered/virtual_wall", 1);
```

Comment:
The topic name should be `/createN/virtual_wall` and the original topic should be `/createN/virtual_wall/raw`.